localstack/services/apigateway/invocations.py

```diff
@@ -215,14 +217,28 @@ def apply_response_parameters(invocation_context: ApiInvocationContext):
     int_responses = integration.get("integrationResponses") or {}
     if not int_responses:
         return response
+
+    integration_type_orig = integration.get("type") or integration.get("integrationType") or ""
+    integration_type = integration_type_orig.upper()
+    if integration_type == "AWS_PROXY":
+        LOG.warning("AWS_PROXY integration type should not apply response parameters")
+        return response
+
     entries = list(int_responses.keys())
     return_code = str(response.status_code)
-    if return_code not in entries:
-        if len(entries) > 1:
-            LOG.info("Found multiple integration response status codes: %s", entries)
-            return response
-        return_code = entries[0]
-    response_params = int_responses[return_code].get("responseParameters", {})
+    # Selecting the right integration response
+    # FIXME is this check valid? It should map to the default one or based on the selectionPattern
+    # Not sure if it gets mapped automatically to a method response if it matches the statusCode by chance
+    LOG.debug(
+        "Found multiple integration response status codes: %s, choosing by selection pattern",
+        entries,
+    )
```

Comment:
question: there could be only one integration response here, right, we're not checking the length of `entries` anymore? Do we still need this logging statement, or the logging in `select_integration_response` sufficient?

Reply:
It can be removed the new function picks up if there is more than one and is also logging.

Reply:
Yes, I just plain up forgot about that!